src/ok.rs

```diff
@@ -25,6 +26,26 @@ struct TestResult {
     pub msg: String,
 }
 
+impl CpuId {
+    pub fn execute(&self) -> TestResult {
+        let res = (self.func)(unsafe { x86_64::__cpuid(self.leaf) });
+        let success = res.0;
+        let msg = if success {
+            if let Some(m) = res.1 {
+                m
+            } else {
+                format!("{}", "Supported".green())
+            }
+        } else {
+            format!("{}", "Unsupported".red())
+        };
```

Comment:
Un-tested and unformatted:

```suggestion
        let (passed, msg) = res;
        let msg = msg.unwrap_or_else(|| if passed { "Supported".green() } else { "Unsupported".red() });
```